nixos/modules/misc/documentation.nix

```diff
@@ -36,6 +60,43 @@ let
           )
           pkgSet;
       in scrubbedEval.options;
+    baseOptionsJSON =
+      let
+        filter =
+          builtins.filterSource
+            (n: t:
+              (t == "directory" -> baseNameOf n != "tests")
+              && (t == "file" -> hasSuffix ".nix" n)
+            );
+      in
+        pkgs.runCommand "lazy-options.json" {
+          libPath = filter "${toString pkgs.path}/lib";
+          pkgsLibPath = filter "${toString pkgs.path}/pkgs/pkgs-lib";
+          nixosPath = filter "${toString pkgs.path}/nixos";
+          modules = map (p: ''"${removePrefix "${modulesPath}/" (toString p)}"'') docModules.lazy;
+        } ''
+          export NIX_STORE_DIR=$TMPDIR/store
+          export NIX_STATE_DIR=$TMPDIR/state
+          ${pkgs.nix}/bin/nix-instantiate \
+            --show-trace \
+            --eval --json --strict \
+            --argstr libPath "$libPath" \
+            --argstr pkgsLibPath "$pkgsLibPath" \
+            --argstr nixosPath "$nixosPath" \
+            --arg modules "[ $modules ]" \
+            --argstr stateVersion "${options.system.stateVersion.default}" \
+            --argstr release "${config.system.nixos.release}" \
+            ${./eval-cacheable-options.nix} > $out \
```

Comment:
This adds this file to the store independently, but it is already in `${toString pkgs.path}` so you could use that instead.

Reply:
nope, that's *much* more expensive than the many small adds this does. :( would be nice if there was a way to tell nix to reuse an existing store path, much like it does for derivations.

Reply:
If you reuse `$nixosPath` you don't need to add anything extra to the store, right?

Reply:
ah, think i misunderstood. yeah, we can totally use `$nixosPath/modules/misc/eval-cacheable-options.nix` instead!